src/test/java/com/exactpro/epfast/template/assertion/InstrEqualityChecker.java

```diff
@@ -0,0 +1,146 @@
+package com.exactpro.epfast.template.assertion;
+
+import com.exactpro.epfast.template.*;
+
+import java.util.List;
+
+import static com.exactpro.epfast.template.assertion.OpEqualityChecker.areSameOperators;
+import static com.exactpro.epfast.template.assertion.TemplateAssert.areSameIdentities;
+import static com.exactpro.epfast.template.assertion.TemplateAssert.areSameReferences;
+
+public class InstrEqualityChecker {
+
+    static boolean areSameInstructions(List<? extends Instruction> actual, List<? extends Instruction> instr) {
+        if (actual == instr) {
+            return true;
+        }
+        if (actual.size() != instr.size()) {
+            return false;
+        }
+
+        for (int i = 0; i < actual.size(); i++) {
+
+            Instruction actualItem = actual.get(i);
+            Instruction instrItem = instr.get(i);
+            if (actualItem != instrItem) {
+                if (!actualItem.getClass().equals(instrItem.getClass())) {
```

Comment:
I just realized it's wrong check. The implementation classes can be different, but both should implement same interface. I think this check should be removed. And checks in if should be changed to
```java
if ((actualItem instanceof TemplateRef) && (instItem instanceof TemplateRef)) {
```